src/operator/tensor/ordering_op-inl.h

```diff
@@ -420,8 +420,6 @@ void TopKImpl(const RunContext &ctx,
     mxnet::op::SortByKeyWorkspaceSize<int, int, xpu>(src.Size()));
   temp_size = std::max(temp_size,
     mxnet::op::SortByKeyWorkspaceSize<int, DType, xpu>(src.Size()));
-  temp_size = std::max(temp_size,
```

Comment:
there was this - https://github.com/apache/incubator-mxnet/pull/14912#pullrequestreview-237632437 concern here by @sxjscience

Reply:
I think this line is to make sure that the code will be correct no matter how we change the `SortByKeyWorkspaceSize` function. For example, if we later change our `SortByKeyWorkspaceSize` as something like:
```
mxnet::op::SortByKeyWorkspaceSize<int, DType, xpu>() --> Return 0
mxnet::op::SortByKeyWorkspaceSize<DType, int, xpu>() --> Return 1000000
```
The current revision will raise an error because the `temp_size` is not correctly updated. I suggest keeping this line to avoid issues in the future.

Reply:
Okay. I am not sure why, but without removing these lines, compiling the code with cuda enabled and fp16 enabled fails.

Reply:
Please try to locate the source of the error. Removing the line does not solve the problem.